### PR TITLE
Pick also bad channels when selecting T1T2 mag indices

### PR DESCRIPTION
**Pull request by mattitoi**
If exclude="bads" (default),
channels marked as bad will not be included.
This will lead to coil type not being fixed
in call to mne.preprocessing.maxwell_filter
3022 (FIFFV_COIL_VV_MAG_T1)to 3024 (FIFFV_COIL_VV_MAG_T3).

#### What does this implement/fix?

`mne.preprocessing.maxwell_filter` fixes the coil type of magnetometers from `3022 (FIFFV_COIL_VV_MAG_T1)` to `3024 (FIFFV_COIL_VV_MAG_T3)`. However, the coil type is not fixed for channels marked as bad. This is because `exclude` argument is `bads` by default for `pick_types`-function.

#### Additional information

Tests run only for `mne/preprocessing/tests/test_maxwell.py`.

```bash
>pytest mne/preprocessing/tests/test_maxwell.py
=========================================================================== test session starts ===========================================================================
platform win32 -- Python 3.10.4, pytest-7.1.1, pluggy-1.0.0
rootdir: C:\Users\matti.toivonen\src\mne-python, configfile: setup.cfg
plugins: cov-3.0.0, harvest-1.10.3, timeout-2.1.0
collected 42 items

mne\preprocessing\tests\test_maxwell.py ..........................................                                                                                   [100%]


========================================================================= slowest 20 test modules =========================================================================
99.98s total   mne/preprocessing/tests/test_maxwell.py


---------------------------------------------- generated xml file: C:\Users\matti.toivonen\src\mne-python\junit-results.xml -----------------------------------------------
========================================================================== slowest 20 durations ===========================================================================
17.04s call     mne/preprocessing/tests/test_maxwell.py::test_shielding_factor
16.42s call     mne/preprocessing/tests/test_maxwell.py::test_spatiotemporal
13.69s call     mne/preprocessing/tests/test_maxwell.py::test_movement_compensation
8.12s call     mne/preprocessing/tests/test_maxwell.py::test_find_bads_maxwell_flat
4.25s call     mne/preprocessing/tests/test_maxwell.py::test_triux
4.12s call     mne/preprocessing/tests/test_maxwell.py::test_all
3.68s call     mne/preprocessing/tests/test_maxwell.py::test_find_bad_channels_maxwell[C:\\Users\\matti.toivonen\\mne_data\\MNE-testing-data\\CTF\\testdata_ctf.ds-bads2-False-False-False-want_bads2-False-None-raw]
3.57s call     mne/preprocessing/tests/test_maxwell.py::test_spatiotemporal_only
2.74s call     mne/preprocessing/tests/test_maxwell.py::test_other_systems
1.92s call     mne/preprocessing/tests/test_maxwell.py::test_basic
1.43s call     mne/preprocessing/tests/test_maxwell.py::test_find_bad_channels_maxwell[C:\\Users\\matti.toivonen\\mne_data\\MNE-testing-data\\MEG\\sample\\sample_audvis_trunc_raw.fif-bads0-False-False-False-want_bads0-False-None-raw]
1.37s call     mne/preprocessing/tests/test_maxwell.py::test_find_bad_channels_maxwell[C:\\Users\\matti.toivonen\\mne_data\\MNE-testing-data\\CTF\\testdata_ctf.ds-bads3-False-False-True-want_bads3-False-None-raw]
1.36s call     mne/preprocessing/tests/test_maxwell.py::test_find_bad_channels_maxwell[C:\\Users\\matti.toivonen\\mne_data\\MNE-testing-data\\MEG\\sample\\sample_audvis_trunc_raw.fif-bads5-True-True-False-want_bads5-True-50-None]
1.28s call     mne/preprocessing/tests/test_maxwell.py::test_find_bad_channels_maxwell[C:\\Users\\matti.toivonen\\mne_data\\MNE-testing-data\\MEG\\sample\\sample_audvis_trunc_raw.fif-bads4-True-True-False-want_bads4-True-50-raw]
1.25s call     mne/preprocessing/tests/test_maxwell.py::test_find_bad_channels_maxwell[C:\\Users\\matti.toivonen\\mne_data\\MNE-testing-data\\MEG\\sample\\sample_audvis_trunc_raw.fif-bads1-True-True-False-want_bads1-False-None-raw]
1.18s call     mne/preprocessing/tests/test_maxwell.py::test_cross_talk
1.14s call     mne/preprocessing/tests/test_maxwell.py::test_esss[bads0-in]
1.12s call     mne/preprocessing/tests/test_maxwell.py::test_fine_calibration
1.10s call     mne/preprocessing/tests/test_maxwell.py::test_mf_skips
1.08s call     mne/preprocessing/tests/test_maxwell.py::test_esss[bads1-in]
===================================================================== 42 passed in 100.96s (0:01:40) ======================================================================

```

### CODE REVIEW

**Comment by agramfort** [comment]
@mattitoi can you add or update a test ? thx 🙏

**Comment by mattitoi** [comment]
@agramfort Updated test for [`fix_mag_coil_types` (which uses is the only function calling this one) via `Raw`](https://github.com/mne-tools/mne-python/blob/575193c1b189733071a0623bf436c08daeab671c/mne/io/fiff/tests/test_raw_fiff.py#L121).

**Comment by mattitoi** [comment]
Multiple tests fail, but they failed even before these changes:

```bash
pytest mne\io\fiff\tests\test_raw_fiff.py
=========================================================================================================== test session starts ============================================================================================================
platform win32 -- Python 3.10.4, pytest-7.1.1, pluggy-1.0.0
rootdir: C:\Users\matti.toivonen\src\mne-python, configfile: setup.cfg
plugins: cov-3.0.0, harvest-1.10.3, timeout-2.1.0
collected 67 items

mne\io\fiff\tests\test_raw_fiff.py ..............................................s.......FF..FF....s..                                                                                                                                [100%]
```

The failures occur in other test-functions and share the following error message:

```bash
______________________________________________________________________________________________________ test_file_like[True-str-file] _______________________________________________________________________________________________________
mne\io\fiff\tests\test_raw_fiff.py:1726: in test_file_like
    _test_raw_reader(read_raw_fif, **kwargs)
mne\io\tests\test_raw.py:413: in _test_raw_reader
    _test_raw_crop(reader, t_prop, kwargs)
mne\io\tests\test_raw.py:423: in _test_raw_crop
    raw_1 = reader(**kwargs)
mne\io\fiff\raw.py:482: in read_raw_fif
    return Raw(fname=fname, allow_maxshield=allow_maxshield,
<decorator-gen-241>:12: in __init__
    ???
mne\io\fiff\raw.py:125: in __init__
    self._preload_data(preload)
mne\io\base.py:566: in _preload_data
    self._data = self._read_segment(
<decorator-gen-205>:12: in _read_segment
    ???
mne\io\base.py:394: in _read_segment
    data = _allocate_data(data_buffer, data_shape, dtype)
mne\io\base.py:2044: in _allocate_data
    data = np.memmap(str(preload), mode='w+', dtype=dtype, shape=shape)
..\..\Miniconda3\envs\mnedev\lib\site-packages\numpy\core\memmap.py:228: in __new__
    f_ctx = open(os_fspath(filename), ('r' if mode == 'c' else mode)+'b')
E   OSError: [Errno 22] Invalid argument: 'C:\\Users\\matti.toivonen\\AppData\\Local\\Temp\\pytest-of-matti.toivonen\\pytest-4\\test_file_like_True_str_file_0\\memmap'
```

File: mne/io/fiff/tests/test_raw_fiff.py [hunk 1]
```diff
@@ -107,10 +107,14 @@ def test_acq_skip(tmp_path):
 
 def test_fix_types():
     """Test fixing of channel types."""
-    for fname, change in ((hp_fif_fname, True), (test_fif_fname, False),
-                          (ctf_fname, False)):
+    for fname, change, bads in (
+        (hp_fif_fname, True, ["MEG0111"]),
```
Comment (mattitoi) [inline mne/io/fiff/tests/test_raw_fiff.py:111]: Need to mark at least one channel as bad to see that its coil type also changed.

**Comment by mattitoi** [comment]
@larsoner Could you check that I got the syntax right with the additions to [`doc/changes/latest.inc` and `doc/changes/names.inc`](https://github.com/mne-tools/mne-python/pull/10639/files#diff-bc201b970a711e7a23731329d2a1a973b73d53f835406d69e250e562be66f390)

**Comment by larsoner** [comment]
No need to check manually when the CIs complain already :)
https://circleci.com/gh/mne-tools/mne-python/44738?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
```
changes/latest.inc:77: WARNING: Mismatch: both interpreted text role prefix and reference suffix.
```
I'll see if I can spot the problem and push a quick commit to fix it

File: doc/changes/latest.inc [hunk 2]
```diff
@@ -70,6 +70,9 @@ Enhancements
 
 Bugs
 ~~~~
+
+- Pick also bad channels when selecting T1T2 mag indices (:gh:`10639` by :newcontrib:`Matti Toivonen`_)
```
Comment (larsoner) [inline doc/changes/latest.inc:74]: Might as well give a hint where people will see the change while fixing the ref bug
```suggestion
- Pick also bad channels when selecting T1T2 magnetometers in :func:`mne.preprocessing.maxwell_filter` (:gh:`10639` by :newcontrib:`Matti Toivonen`)
```

**Comment by larsoner** [comment]
All better 

https://output.circle-artifacts.com/output/job/4a7b3a20-0117-4f67-94f1-2f31f8c67d29/artifacts/0/dev/whats_new.html#bugs

Will merge once CIs come back happy. Thanks in advance @mattitoi !

**Pull request merged**